core/sql/sqlcomp/CmpSeabaseDDLroutine.cpp

```diff
@@ -995,180 +1561,293 @@ void CmpSeabaseDDL::createSeabaseRoutine(
+      if ( specifiedSig.length() == 0 )
+        optionalSig = NULL;
+      else
+        optionalSig = (char *)specifiedSig.data();
      
-     ComBoolean isJavaMain =
-       ((str_cmp_ne(createRoutineNode->getJavaMethodName(), "main") == 0) ? TRUE : FALSE);
-
-     LmResult createSigResult;
-     LmJavaSignature *lmSignature =  new (STMTHEAP) LmJavaSignature(NULL,
-                                                                    STMTHEAP);
-     createSigResult = lmSignature->createSig(paramType, subType, direction,
-                                              numParams, COM_UNKNOWN_FSDT, 0,
-                                              createRoutineNode->getMaxResults(), optionalSig, isJavaMain, sigBuf,
-                                              MAX_SIGNATURE_LENGTH,
-                                              CmpCommon::diags());
-     NADELETE(lmSignature, LmJavaSignature, STMTHEAP);
-     delete [] paramType;
-     delete [] subType;
-     delete [] direction;
-
-     // Lm returned error. Lm fills diags area, so no need to worry about diags.
-     if (createSigResult == LM_ERR)
-     {
-       *CmpCommon::diags() << DgSqlCode(-1231)
-			  << DgString0(extRoutineName);
-       deallocEHI(ehi); 
-       processReturn();
-       return;
-     }
+      ComBoolean isJavaMain =
+        ((str_cmp_ne(createRoutineNode->getJavaMethodName(), "main") == 0) ? TRUE : FALSE);
+
+      LmResult createSigResult;
+      LmJavaSignature *lmSignature =  new (STMTHEAP) LmJavaSignature(NULL,
+                                                                     STMTHEAP);
+      createSigResult = lmSignature->createSig(paramType, subType, direction,
+                                               numParams, COM_UNKNOWN_FSDT, 0,
+                                               createRoutineNode->getMaxResults(), optionalSig, isJavaMain, sigBuf,
+                                               MAX_SIGNATURE_LENGTH,
+                                               CmpCommon::diags());
+      NADELETE(lmSignature, LmJavaSignature, STMTHEAP);
+      delete [] paramType;
+      delete [] subType;
+      delete [] direction;
+
+      // Lm returned error. Lm fills diags area, so no need to worry about diags.
+      if (createSigResult == LM_ERR)
+        {
+          *CmpCommon::diags() << DgSqlCode(-1231)
+                              << DgString0(extRoutineName);
+          deallocEHI(ehi); 
+          processReturn();
+          return;
+        }
 
-     numJavaParam = (isJavaMain ? 1 : numParams);
+      numJavaParam = (isJavaMain ? 1 : numParams);
+
+      if( libBlobHandle[0] != '\0' )
+        {
+          NAString dummyUser;
+          NAString cachedLibName, cachedLibPath;
+          
+          if(ComGenerateUdrCachedLibName(extPath,redefTime,libSchNamePart,dummyUser, cachedLibName, cachedLibPath))
+            {
+              *CmpCommon::diags() << DgSqlCode(-1231)
+                                  << DgString0(extRoutineName);
```

Comment:
There is a literal for this: CAT_UNABLE_TO_CREATE_ROUTINE.  This is another error that could fail for various reasons and it could be improved to return more details.